### PR TITLE
add e2e: Comments mock structure

### LINKED ISSUE #2708: e2e: Comments mock structure
Part of #2642.

Comments tests are still using the online api approach. We need to migrate it to the new mock structure introduced by (#2624).

We currently don't need new test suites, the main focus here is to first migrate and then code more tests using this new structure.

## Tips and Tricks

- Run only `comments` test file: `yarn cy:run -- --spec "cypress/e2e/comments/comments.js"`
- Setup the API environment for all possible routes reached by the tests suite using a `beforeEach` statement
- Check on politeiawww's logs to check if all requests were mocked properly

### PR DESCRIPTION
**Pull request by vibros68**
Closes #2708

This diff implement mock data for comments on detail proposal page. So
the teste2e process will not submit to the server. The new mock 
structure introduced by #2624.

### CODE REVIEW

**Review by victorgcramos — CHANGES_REQUESTED** [review]
Code looking really good! Clean, readable, simple. Great job!

One thing I noticed is that comments bundle test case is breaking when you run all tests:

75b60a0a-e112-4a72-ba30-287957c3b0b3


Left some inline comments regarding the old mock api usage and docs.

**Review by victorgcramos — CHANGES_REQUESTED** [review]
Docs are good. But I think you should move the `general` support structure to the correct place, in order to match the backend design.

**Comment by vibros68** [comment]
Screenshot from 2022-01-22 01-32-00

I changed commentVotes teste2e to the new structure since it has some problems to be fixed.

**Review by victorgcramos — CHANGES_REQUESTED** [review]
You shouldn't include commentVotes tests here on this PR, because it's not the correct scope. Also, I noticed that you removed the test cases for comments edits, and you should also include them on the new mock structure.

Left some inline comments.

File: teste2e/cypress/support/core/generate.js [hunk 1]
```diff
@@ -33,37 +33,6 @@ export function UserMetadata({ userid, publickey, signature }) {
...
-      reason,
-      signature,
-      status,
-      token,
-      version,
-      timestamp
-    });
-  return new Metadata("usermd", 2, payload);
-}
-
```
[the remark this answers is not in the document]
Comment (vibros68) [reply teste2e/cypress/support/core/generate.js:66]: I added and then I do not use it anymore. But I forgot to remove it on that pr. I am sorry. It is my bad.

File: teste2e/cypress/support/core/commands.js [hunk 2]
```diff
@@ -12,28 +12,36 @@ import { repliers, errorReply, API_BASE_URL } from "./api";
...
-      .intercept(`${baseUrl}/${endpoint}`, (req) => {
-        const replyfn = repliers[endpoint];
+      .intercept(`${path}${search}`, (req) => {
+        const replyfn = repliers[replyIndex];
         if (!replyfn)
           throw new Error(`no ${packageName} replier found for ${endpoint}`);
         const requestParams = req.body || {};
         if (error) {
           req.reply({
+            delay,
```
[the remark this answers is not in the document]
Comment (vibros68) [reply teste2e/cypress/support/core/commands.js:32]: delay should be the config of the request. While responseParams is the param which is received in the form of the request

File: teste2e/cypress/support/core/generate.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (victorgcramos) [reply teste2e/cypress/support/core/generate.js:66]: no problem! That's fine.

**Comment by vibros68** [comment]
Screenshot from 2022-01-27 16-28-14

The last result

**Review by victorgcramos — APPROVED** [review]
LGTM!

**Review by tiagoalvesdulce — APPROVED** [review]
ACK

**Pull request merged**